docs/test-framework-dev-guide.md

```diff
@@ -49,6 +49,41 @@ pass `[testName]` to `go test` as `--run=[testName]`.
 
 - `mage integration:matrix` to run all tests on the complete matrix of supported operating systems and architectures of the Elastic Agent.
 
+#### Passing additional go test flags
+
+When running the tests we can pass additional go test flag using the env variable `GOTEST_FLAGS`.
+
+These flags are passed also when calculating batches for remote execution of integration tests.
+This allows for selecting a subset of test in a convenient way (see examples below)
+
+This feature is intended mostly for integration tests debugging/development without the need for
+new mage targets corresponding to a new set of test flags.
+
+A few examples:
+
+##### Run a single test with an exact match
+We want to run only the test named "TestStandaloneUpgrade"
+`GOTEST_FLAGS="-test.run ^TestStandaloneUpgrade$" mage integration:test`
+
+##### Run a tests matching a partial expression
+We want to run any test with "Upgrade" in the name
+`GOTEST_FLAGS="-test.run Upgrade" mage integration:test`
+##### Run a single test and signal that we want the short version
+We pass a `-test.short` flag along with the name match
+`GOTEST_FLAGS="-test.run ^TestStandaloneUpgrade$ -test.short" mage integration:test`
+
+##### Run a single test multiple times
+We pass a `-test.count` flag along with the name match
+`GOTEST_FLAGS="-test.run ^TestStandaloneUpgrade$ -test.count 10" mage integration:test`
+
+##### Run specific tests
+We pass a `-test.run` flag along with the names of the tests we want to run in OR
+`GOTEST_FLAGS="-test.run ^(TestStandaloneUpgrade|TestFleetManagedUpgrade)$" mage integration:test`
+
+##### Limitations
+Due to the way the parameters are passed to `devtools.GoTest` the value of the environment variable
+is split on space, so not all combination of flags and their values may be correctly split.
```

Comment:
Thanks for adding this section.

Does this limitation have something to do with some go test flags taking arguments, e.g. `-test.run <test name regex>`, vs. others not taking arguments, e.g. `-short`?

Reply:
It does not have any problems with flags that take arguments or not... 
The problem is that when one argument contains a space, it will be split incorrectly.
The correct way would be to:
- either reimplement the flag parser of `go test` (those are not exposed sadly and reimplementing it and keeping it in sync with the future versions of go it will be probably too much work)
- or define a completely different set of flags and translate them into go test flags (kinda misses the point of the PR)

Another alternative is to build the `go test` command differently so that we can plonk the flags as a single string being completely agnostic about the contents but that is also a bit of work...

For this PR I went with the simple and stupid solution and I couldn't think of a scenario where I needed a smarter parsing of flags: we'll cross that bridge when we have to.

Reply:
Understood, thanks for clarifying.  I'm good with the simple solution for now.